.gitignore

```diff
@@ -8,3 +8,4 @@ build/
 packages
 .idea/
 
+pubspec.lock
```

Comment:
For an app it makes sense so that the configuration is the same on every machine.

For a library, you WANT to support many versions of dependencies.